### PR TITLE
chore: introduce `JsonT`

### PR DESCRIPTION
**Pull request by ssddOnTop**

### CODE REVIEW

File: src/core/json/jsont.rs [hunk 1]
```diff
@@ -0,0 +1,102 @@
+#![allow(unused)]
+
+use std::collections::HashMap;
+
+pub trait JsonT {
+    type Output;
+    type Input;
+
+    fn array_ok(value: &Self::Input) -> Result<&Vec<Self::Output>, &str>;
```
Comment (tusharmath) [inline src/core/json/jsont.rs:9]: Return an `Option`

File: src/core/json/jsont.rs [hunk 2]
```diff
@@ -0,0 +1,102 @@
+#![allow(unused)]
+
+use std::collections::HashMap;
+
+pub trait JsonT {
+    type Output;
+    type Input;
```
Comment (tusharmath) [inline src/core/json/jsont.rs:7]: Use a type called `Json` for both `Input` and `Output`.

File: src/core/json/jsont.rs [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+    fn bool_ok(value: &Self::Input) -> Result<bool, &str>;
+    fn null_ok(value: &Self::Input) -> Result<(), &str>;
+    fn option_ok(value: &Self::Input) -> Result<Option<&Self::Output>, &str>;
+    fn get_path<'a, T: AsRef<str>>(value: &'a Self::Input, path: &'a [T]) -> Option<&'a Self::Output>;
+    fn get_key<'a>(value: &'a Self::Input, path: &'a str) -> Option<&'a Self::Output>;
+    fn new(value: &Self::Output) -> &Self;
+    fn group_by<'a>(value: &'a Self::Input, path: &'a [String]) -> HashMap<String, Vec<&'a Self::Output>>;
+}
+
+impl JsonT for serde_json::Value {
```
Comment (tusharmath) [inline src/core/json/jsont.rs:23]:
```suggestion
impl JsonT for JsonTSerde {
```

We need a new type here.

File: src/core/json/jsont.rs (hunk 2 above)
Comment (ssddOnTop) [reply src/core/json/jsont.rs:7]: I am thinking of using `JsonT` directly

Comment (ssddOnTop) [reply src/core/json/jsont.rs:7]: `type Output: JsonT;`

File: src/core/json/jsont.rs [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+    use super::JsonT;
+    use serde_json::Value;
+
+    #[test]
+    fn test_array_ok() {
+        let value = Value::Array(vec![Value::Null]);
+        let result = <Value as JsonT>::array_ok(&value);
+
+        assert_eq!(result, Ok(&vec![Value::Null]));
+    }
```
Comment (tusharmath) [inline src/core/json/jsont.rs:101]: copy tests from JsonLike.

File: src/core/json/jsont.rs [hunk 5]
```diff
@@ -0,0 +1,102 @@
+#![allow(unused)]
+
+use std::collections::HashMap;
+
+pub trait JsonT {
```
Comment (tusharmath) [inline src/core/json/jsont.rs:5]: add some docs :)

File: src/core/json/jsont.rs (hunk 3 above)
Comment (ssddOnTop) [reply src/core/json/jsont.rs:22]: marking this as resolved

**Pull request closed without merging**